service/src/main/java/com/ge/predix/acs/service/policy/validation/PolicySetValidatorImpl.java

```diff
@@ -83,10 +87,70 @@ public void removeCachedConditions(final PolicySet policySet) {
     @Override
     public void validatePolicySet(final PolicySet policySet) {
         validateSchema(policySet);
+        List<String> obligationIds = validateObligationsAndGetIds(policySet.getObligations());
         for (Policy p : policySet.getPolicies()) {
             validatePolicyConditions(p.getConditions());
             validatePolicyActions(p);
+            vaLidatePolicyObligations(obligationIds, p);
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    private List<String> validateObligationsAndGetIds(final List<Obligation> obligations) {
+        List<String> ids = new ArrayList<>();
+        List<String> repeatedIds = new ArrayList<>();
+        int index = 0;
+
+        for (Obligation obligation : obligations) {
+            if (StringUtils.isEmpty(obligation.getId())) {
```

Comment:
That requirement can be defined as part of JSON schema (@JsonProperty(required=true)) and validated as part of JSON schema validation.  Also add a unit test.